cms/serializers.py

```diff
@@ -30,11 +32,42 @@ def get_financial_assistance_form_url(self, instance):
         """
         Returns URL of the Financial Assistance Form.
         """
-        financial_assistance_page = (
-            instance.get_children().type(FlexiblePricingRequestForm).live().first()
-        )
+        financial_assistance_page = None
+        base_url = None
+        if instance.product.program:
+            print("CP", instance.product.program)
+            program_page = ProgramPage.objects.filter(
+                program_id=instance.product.program
+            ).first()
+            base_url = program_page.get_url()
+
+            financial_assistance_page = FlexiblePricingRequestForm.objects.filter(
+                selected_program=instance.product.program
+            ).first()
+            if financial_assistance_page is None:
+                financial_assistance_page = (
+                    program_page.get_children()
+                    .type(FlexiblePricingRequestForm)
+                    .live()
+                    .first()
+                )
+
+        if financial_assistance_page is None:
+            base_url = instance.get_url()
+            financial_assistance_page = FlexiblePricingRequestForm.objects.filter(
+                selected_course=instance.product
+            ).first()
+
+        if financial_assistance_page is None:
+            base_url = instance.get_url()
+            financial_assistance_page = (
+                instance.get_children().type(FlexiblePricingRequestForm).live().first()
+            )
+
+        print("CP", " 64 ", financial_assistance_page)
+
         return (
-            f"{instance.get_url()}{financial_assistance_page.slug}/"
+            f"{base_url}{financial_assistance_page.slug}/"
```

Comment:
The Wagtail `get_url` method should probably be used here on the request form page. It should return the URL with the special formatting we use for course/program pages (which use the readable ID rather than the page slug).